src/tiled/mainwindow.cpp

```diff
@@ -1509,6 +1497,99 @@ void MainWindow::retranslateUi()
     CommandManager::instance()->retranslateUi();
 }
 
+void MainWindow::exportMapAs(MapDocument *mapDocument)
+{
+    QString fileName = mapDocument->fileName();
+    QString selectedFilter =
+            mSettings.value(QLatin1String("lastUsedExportFilter")).toString();
+    auto exportDetails = chooseExportDetails<MapFormat>(fileName,
+                                                        mapDocument->lastExportFileName(),
+                                                        selectedFilter,
+                                                        this);
+    if (!exportDetails.valid()) {
+        return;
+    }
+
+    // Check if writer will overwrite existing files here because some writers
+    // could save to multiple files at the same time. For example CSV saves
+    // each layer into a separate file.
+    QStringList outputFiles = exportDetails.mFormat->outputFiles(mapDocument->map(), exportDetails.mFileName);
```

Comment:
This `outputFiles` call and related overwrite check should technically also by applied when exporting tilesets, even though right now we know that there are no tileset formats that export multiple files. The code could just be moved into `chooseExportDetails`, right?

Reply:
This section relies on ``MapFormat::outputFiles`` and this isn't currently a part of ``TilesetFormat`` which is why I separated it from ``chooseExportDetails``, if it was then it could easily be included.

What would you like done here?

Reply:
Sorry, I didn't realize `outputFiles` was only part of the `MapFormat`. I see in the current code it was part of `FileFormat`, but that didn't make much sense since it receives a map as parameter. So it's good that you moved it to `MapFormat` and then you can disregard this comment. :-)